library.go

```diff
@@ -1827,6 +1826,49 @@ func string_split(fnname string, recv_ Value, args Tuple, kwargs []Tuple) (Value
 	return NewList(list), nil
 }
 
+func reverse(s string, size int) string {
+	rev := make([]rune, 0, size)
+
+	for _, point := range s {
+		rev = append(rev, point)
+	}
+
+	for i := 0; i < size/2; i++ {
+		rev[i], rev[size-1-i] = rev[size-1-i], rev[i]
+	}
+
+	return string(rev)
+}
+
+func rsplit(s string, max int) []string {
+	size := utf8.RuneCountInString(s)
+	splitCount := 0   // Our current number of splits made.
+	currIndex := size // A backward working index.
+	lastStop := size  // After we make a split, advance this back.
+	parts := make([]string, 0, max)
```

Comment:
Could you add a test for max < 0?

Reply:
Heh, I forgot you could pass -1 to `rsplit` to signify no limits. I will just remove the capacity in this `make` statement.

Reply:
I just realized that the caller handles the zero and negative cases, thus your function is guaranteed  `max > 0`.  You can ignore my comment.